include/glow/Optimizer/Partition.h

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_OPTIMIZER_PARTITION_H
+#define GLOW_OPTIMIZER_PARTITION_H
+
+#include "glow/Graph/Graph.h"
+
+#include <llvm/ADT/DenseMap.h>
+
+namespace glow {
+
+/// Data structure representing a set of interdependent Functions and the
+/// dependences between them.
+class FunctionGraph {
+  /// Dependency map type.
```

Comment:
Could you expand the comment here to something like "Maps a function to the set of functions it depends on. The execution of a function cannot start until all of its dependencies are executed."